skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -34,12 +50,65 @@ public Page<EndorsementModel> getEndorsements(PageRequest pageRequest) {
         return endorsementRepository.findAll(pageRequest);
     }
 
+    /* TODO:Dummy JSON code, needs to be changed as part of #103 */
     @Override
     public Page<EndorsementModelFromJSON> getEndorsementsFromDummyData(
             PageRequest pageRequest, String skillID, String userID) throws IOException {
 
-        // TODO: temporary stub, implement in followup PR
-        return null;
+        try {
+            List<EndorsementModelFromJSON> endorsementModelFromJSONList = readEndorsementsFromJSON();
+
+            if (endorsementModelFromJSONList.isEmpty()) {
+                return Page.empty(pageRequest);
+            }
+
+            List<EndorsementModelFromJSON> filteredEndorsements =
+                    filterEndorsements(endorsementModelFromJSONList, skillID, userID);
+            return createPagedEndorsements(filteredEndorsements, pageRequest);
+        } catch (IOException e) {
+            logger.error("Error reading endorsements JSON data: {}", e.getMessage());
+            throw new IOException("Error reading endorsements JSON data", e);
+        }
+    }
+
+    private List<EndorsementModelFromJSON> readEndorsementsFromJSON() throws IOException {
+        ClassPathResource resource = new ClassPathResource(dummyEndorsementDataPath);
+        return objectMapper.readValue(
+                resource.getInputStream(), new TypeReference<List<EndorsementModelFromJSON>>() {});
+    }
+
+    private List<EndorsementModelFromJSON> filterEndorsements(
+            List<EndorsementModelFromJSON> endorsements, String skillID, String userID) {
+        return endorsements.stream()
+                .filter(endorsement -> isMatchingSkillId(endorsement, skillID))
+                .filter(endorsement -> isMatchingUserId(endorsement, userID))
+                .collect(Collectors.toList());
```

Comment:
Hi @akhilkh2000 
Here, we want to collect only those endorsements that have matching userId or skillId, right? But, if an endorsement does not match the skillId, will it still check for userId match?

Reply:
So if we’ve passed skillID = <some UUID that does not exist in our dummy data> and userID = <some UUID that exists> it will return empty because query parameters by nature are an AND. So if I pass a specific value for either of them that does not exist, it will not match any.